injected/unit-test/page-context-dom.spec.js

```diff
@@ -0,0 +1,199 @@
+            settings: { maxLength: 10000, maxDepth: 100, excludeSelectors: null, includeIframes: false, trimBlankLinks: false },
+        },
+        {
+            name: 'mixed-formatting',
+            html: '<p>This has <strong><em>bold and italic</em></strong> together.</p>',
+            settings: defaultSettings,
+        },
+        {
+            name: 'article-structure',
+            html: `<article>
+                <h1>Article Title</h1>
+                <p>By <strong>Author Name</strong></p>
+                <p>This is the introduction paragraph with some <em>emphasis</em>.</p>
+                <h2>First Section</h2>
+                <p>Content of the first section.</p>
+                <ul>
+                    <li>Point one</li>
+                    <li>Point two</li>
+                </ul>
+                <h2>Second Section</h2>
+                <p>Content with a <a href="https://example.com">link</a>.</p>
+            </article>`,
+            settings: defaultSettings,
+        },
+        {
+            name: 'blog-post',
+            html: `<main>
+                <h1>Blog Post Title</h1>
+                <p>Published on January 1, 2024</p>
+                <img src="header.jpg" alt="Header image">
+                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
+                <h2>Key Takeaways</h2>
+                <ol>
+                    <li>First takeaway</li>
+                    <li>Second takeaway</li>
+                    <li>Third takeaway</li>
+                </ol>
+                <p>Read more on <a href="https://blog.example.com">our blog</a>.</p>
+            </main>`,
+            settings: defaultSettings,
+        },
+    ];
+
+    for (const testCase of testCases) {
+        it(`should convert ${testCase.name} to markdown`, () => {
+            // Create a JSDOM instance
+            const dom = new JSDOM(`<!DOCTYPE html><html><body>${testCase.html}</body></html>`);
+            const { window } = dom;
+            const { document } = window;
+
+            // Save original globals
+            const originalWindow = global.window;
+            const originalNode = global.Node;
+
+            // Set up global window and Node for the imported function
+            global.window = window;
+            global.Node = window.Node;
+
+            try {
+                // Convert to markdown
+                const markdown = domToMarkdown(document.body, testCase.settings, 0).trim();
+
+                // Write output file
+                const outputFile = join(outputDir, `${testCase.name}.md`);
+                writeFileSync(outputFile, markdown, 'utf8');
+
+                // Check if expected file exists
+                const expectedFile = join(fixturesDir, 'expected', `${testCase.name}.md`);
+                if (existsSync(expectedFile)) {
+                    const expected = readFileSync(expectedFile, 'utf8').trim();
+                    expect(markdown).toEqual(expected);
```

Comment:
Why not `expect(markdown).toMatchSnapshot()`?

https://jestjs.io/docs/expect#tomatchsnapshotpropertymatchers-hint

Reply:
Oh my mistake!